core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java

```diff
@@ -210,6 +255,18 @@ private long remaining(long startPollEpochMilli, long maxPollDuration)  {
         return records.isEmpty() ? null : records;
     }
 
+    @Override
+    public void commitRecord(SourceRecord record, RecordMetadata metadata) throws InterruptedException {
```

Comment:
who is throwing InterrupedException ? we need to be sure not to left an exchange uncompleted and not tracked any more in the array

Reply:
done in https://github.com/apache/camel-kafka-connector/pull/1093